### PR TITLE
Shared dispatcher and connection pool

### PR DESCRIPTION
**Pull request by dansanduleac**
Decided to share dispatcher and connection pool for all clients produced by the `OkHttpClients` class. This way we can enforce limits on connections globally, and allow reusing connections to the same host even if they come from different clients.

@uschi2000

### CODE REVIEW

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 1]
```diff
@@ -54,6 +54,16 @@
      */
     private static final ExecutorService executionExecutor = Tracers.wrap(new Dispatcher().executorService());
 
+    /**
+     * Shared dispatcher with static executor service.
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: super nit: can use `/** One-line javadoc. */`

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 2]
```diff
@@ -54,6 +54,16 @@
...
+     */
+    private static final Dispatcher dispatcher = createDispatcher();
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: maybe move the initialization to a `static {}` block so it's clear that the createDispatcher method shouldn't be used?

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 3]
```diff
@@ -212,6 +221,7 @@ private static UserAgent augmentUserAgent(UserAgent agent, Class<?> serviceClass
                 ConnectionSpec.CLEARTEXT);
     }
 
+    @SuppressWarnings("checkstyle:HiddenField")
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: that seems avoidable

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java (hunk 1 above)
Comment (dansanduleac) [reply okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: sure. I just followed the style of the above field's javadoc, which admittedly just barely doesn't fit on 1 line.

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**